### PR TITLE
MES-3617 create rekey reason components

### PR DESCRIPTION
**Pull request by buzzub**
## Description
Move rekey reason fields into separate components. A future PR will refactor the rekey reason page to use these.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by MarkPatt89 — APPROVED** [review]

File: src/pages/rekey-reason/components/other-reason/other-reason.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  reasonTextChanged(reason: string): void {
+    this.reasonChange.emit(reason);
+  }
+
+  characterCountChanged(charactersRemaining: number) {
+    this.reasonDescriptionCharsRemaining = charactersRemaining;
+  }
+
+  getCharacterCountText() {
```
Comment (TomBillingtonUK) [inline src/pages/rekey-reason/components/other-reason/other-reason.ts:70]: May be me not understanding the character counter, but doesn't the directive do all of this stuff

File: src/pages/rekey-reason/components/other-reason/other-reason.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  <ion-col col-40>
+    <input id="otherSelected" [checked]="selected" type="checkbox" class="gds-checkbox"
+      formControlName="otherSelected" (change)="selectedValueChanged($event.target.checked)">
+    <label for="otherSelected">
+      Other (specify reason)
+    </label>
+  </ion-col>
+  <ion-col>
+    <div class="selected-border" id="reasonSection" *ngIf="selected">
+      <textarea id="otherReason" formControlName="reason" charLimit="200" charCount
```
Comment (TomBillingtonUK) [inline src/pages/rekey-reason/components/other-reason/other-reason.html:12]: This doesn't look like it's using the directive for this

**Review by TomBillingtonUK — APPROVED** [review]
Looks good - Added comments of where a future refactor could be done , if it's in this pr or another it doesn't matter :)

File: src/pages/rekey-reason/components/other-reason/other-reason.ts (hunk 1 above)
Comment (buzzub) [reply src/pages/rekey-reason/components/other-reason/other-reason.ts:70]: The directive emits the count, but currently the job of displaying the count and validation message is manual. There's work here for a future refactor.

**Pull request merged**